src/index.ts

```diff
@@ -184,6 +186,185 @@ export function HTPipeAttachData<
   }
 }
 
+// left has preAuthorize and right has preAuthorize
+export function HTPipePreAuthorize<
+  TLeft extends HasPreAuthorize<
+    any,
+    | boolean
+    | (TRight extends HasPreAuthorize<any, any>
+        ? Pick<
+            Parameters<TRight['preAuthorize']>[0],
+            keyof ReturnType<
+              TLeft extends HasPreAuthorize<any, any>
+                ? TLeft['preAuthorize']
+                : () => {}
+            >
+          >
+        : any)
+  >,
+  TRight extends HasPreAuthorize<any, any>,
+  TContextInLeft extends Parameters<TLeft['preAuthorize']>[0],
+  TContextInRight extends Parameters<TRight['preAuthorize']>[0],
+  TContextOutLeft extends ReturnType<TLeft['preAuthorize']>,
+  TContextOutRight extends ReturnType<TRight['preAuthorize']>
+>(
+  left: TLeft,
+  right: TRight
+): HasPreAuthorize<
```

Comment:
keyof TContextOutLeft
?
... what if TContextOutLeft is actually boolean?
- [ ] fix return value "keyof" like line 198

Reply:
done